src/Voron/StorageEnvironment.cs

```diff
@@ -125,6 +128,7 @@ public StorageEnvironment(StorageEnvironmentOptions options)
             try
             {
                 SelfReference.Owner = this;
```

Comment:
Huh? You have a strong refence here, no?

Reply:
https://stackoverflow.com/questions/8489241/how-does-garbage-collection-collect-self-referential-objects